Assets/HoloToolkit/Input/Scripts/InputManager.cs

```diff
@@ -228,30 +235,35 @@ public void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.EventFunction<
             }
 
             // Use focused object when OverrideFocusedObject is null.
-            GameObject focusedObject = (OverrideFocusedObject == null) ? GazeManager.Instance.HitObject : OverrideFocusedObject;
+            GameObject focusedObject = OverrideFocusedObject ?? GazeManager.Instance.HitObject;
+            GameObject modalInput = modalInputStack.Count > 0 ? modalInputStack.Peek() : null;
+            GameObject fallbackInput = fallbackInputStack.Count > 0 ? fallbackInputStack.Peek() : null;
 
-            // Send the event to global listeners
+            // Send the event to global listeners.
             for (int i = 0; i < globalListeners.Count; i++)
             {
-                // Global listeners should only get events on themselves, as opposed to their hierarchy
-                ExecuteEvents.Execute(globalListeners[i], eventData, eventHandler);
+                // Only execute the global event if it's not already being handled.
```

Comment:
I think the real fix to the input manager is to use the "Used" field of eventData and properly consider it. SOmething like this:

`
GameObject consumingObj = ExecuteEvents.ExecuteHierarchy(priorityObject, eventData, eventHandler);
`
`
if (eventData.used)
{
return;
}
`

This would prevent events from going to every input handler if a handler calls eventData.Use().